homeassistant/components/samsungtv/media_player.py

```diff
@@ -246,10 +248,19 @@ async def async_play_media(self, media_type, media_id, **kwargs):
             await asyncio.sleep(KEY_PRESS_TIMEOUT, self.hass.loop)
         await self.hass.async_add_executor_job(self.send_key, "KEY_ENTER")
 
+    def _wake_on_lan(self):
+        """Wake the device via wake on lan."""
+        send_magic_packet(self._mac, ip_address=self._host)
+        # If the ip address changed since we last saw the device
+        # broadcast a packet as well
+        send_magic_packet(self._mac)
```

Comment:
would assume to broadcast based on mac should be enough to perform WoL :thinking:

Reply:
If they have a bifurcated network directly sending to the ip address may be needed

Reply:
does "bifurcated" include "vpn" ? (this is the day to learn new words 😄

Reply:
magic packets can only be send inside the same/own broadcast domain, so no to vpn, nor only routed reachable targets ;)

But yes to add also ip in the first, to be able to handle multi NIC setups 👍

Reply:
> does "bifurcated" include "vpn" ? (this is the day to learn new words 😄

It _could_ any network interface vpn or otherwise. In this case its likely a second network physical interface.